course7/homeworks/kushagra/hw2/ls_hw2.cpp

```diff
@@ -0,0 +1,169 @@
+		for (fs::directory_entry& x : fs::directory_iterator(dir_path)) {
+			if (fs::is_regular_file(x)) {
+				dir_size += fs::file_size(x);
+			}
+		}
+	}
+	return dir_size;
+}
+
+
+/*Display the directory contents, along with the sizes. The size of the directory is calculated by the sum of all
+the sizes of immediate descendant files in the directory
+*/
+void print_detailed_dir_contents(const fs::path& display_path) {
+	int file_size = 0;
+	int dir_size = get_dir_size(display_path);
+	std::cout << std::to_string(dir_size) << "    " << display_path.filename().generic_string()
+			  << fs::path::preferred_separator << std::endl;
+	for (fs::directory_entry& x : fs::directory_iterator(display_path)) {
+		if (fs::is_regular_file(x.path())) {
+			file_size = fs::file_size(x.path());
+			std::cout << "    " << std::to_string(file_size) << "    " << x.path().filename().string() << std::endl;
+		}
+		else {
+			dir_size = get_dir_size(x.path());
+			std::cout << "    " << std::to_string(dir_size) << "    " << x.path().filename().generic_string()
+					  << fs::path::preferred_separator << std::endl;
+		}
+    }
+}
+
+
+/*Display the directory contents, along with the sizes. The size of the directory is calculated by the sum of all
+the sizes of descendant files in the directory, as well as its subdirectories.
+*/
+void print_detailed_dir_contents(const fs::path& display_path, std::string& spaces) {
+	int dir_size = 0;
+	int file_size = 0;
+	if (fs::is_regular_file(display_path)) {
+		file_size = fs::file_size(display_path);
+		std::cout << spaces << std::to_string(file_size) << "    " << display_path.filename().string() << std::endl;
+	}
+    else {
+		dir_size = get_dir_size(display_path, true);
+        std::cout << spaces << std::to_string(dir_size) << "    " << display_path.filename().generic_string()
+				  << fs::path::preferred_separator << std::endl;
+        spaces += "    ";
+        for (fs::directory_entry& x : fs::directory_iterator(display_path)) {
+            print_detailed_dir_contents(x.path(), spaces);
+        }
+    }
+}
+
+
+int main(int argc, char** argv)
+{
+	std::string path_name = ".";
+	bool details = false, recursive = false;
+	
+    po::options_description desc("Display contents of a directory.");
+    desc.add_options()
+        ("help,h", "Display help text")
+        ("path_name,p", po::value<std::string>()->default_value("."), "Path to display")
+        ("long,l", po::bool_switch()->default_value(false)->implicit_value(true),
+			"Display full details")
+        ("tee,t", po::bool_switch()->default_value(false)->implicit_value(true),
+			"Recursively display directory contents");
+	po::positional_options_description posOption;
+    posOption.add("path_name", -1);
+    po::variables_map vm;
+	po::store(po::command_line_parser(argc, argv).options(desc).positional(posOption).run(), vm);
+	po::notify(vm);
+
+	if(vm.count("path_name")) {
+		path_name = vm["path_name"].as<std::string>();
+		if (path_name.back() == '/') {
```

Comment:
is this really required?

Reply:
Maybe I should try to handle this better - I was getting an extra '/' while printing start directory names, if the input was a path to a directory, and contained a '/' at the end. I chose to remove it like this, but I suppose this maybe an unnecessary check.